File: Swift/LC_Find-the-difference/LC_Find-the-differenceTests/SolutionTests.swift
//
//  SolutionTests.swift
//  LC_Find-the-differenceTests
//
//  Created by Diego Jerez Barroso on 07/02/2021.
//

import XCTest
@testable import LC_Find_the_difference

class SolutionTests: XCTestCase {
    
    private var sut: Solution!
    
    override func setUp() {
        super.setUp()
        sut = Solution()
    }
    
    func test1() {
        
        let s = "abcd"
        let t = "abcde"
        
        XCTAssertEqual(sut.findTheDifference(s, t), "e")
    }
    
    func test2() {
        
        let s = ""
        let t = "y"
        
        XCTAssertEqual(sut.findTheDifference(s, t), "y")
    }
    
    func test3() {
        
        let s = "a"
        let t = "aa"
        
        XCTAssertEqual(sut.findTheDifference(s, t), "a")
    }
    
    func test4() {
        
        let s = "ae"
        let t = "aea"
        
        XCTAssertEqual(sut.findTheDifference(s, t), "a")
    }
    
    func test5() {
        
        let s = "abcdd"
        let t = "abcddd"
        
        XCTAssertEqual(sut.findTheDifference(s, t), "d")
    }
}

File: Python/LC_Find-the-difference/SolutionTests.py
import unittest
from Solution import Solution

class SolutionTests(unittest.TestCase):
    
    def test1(self):
        sut = Solution()
        self.assertEqual(sut.findTheDifference("abcd", "abcde"), "e")

    def test2(self):
        sut = Solution()
        self.assertEqual(sut.findTheDifference("", "y"), "y")

    def test3(self):
        sut = Solution()
        self.assertEqual(sut.findTheDifference("a", "aa"), "a")

    def test3(self):
        sut = Solution()
        self.assertEqual(sut.findTheDifference("abcdd", "abcddd"), "d")

if __name__ == "__main__":
    unittest.main()